bn-apps/memberships-management/membership-service/src/main/kotlin/net/corda/businessnetworks/membership/bno/RevokeMembershipFlow.kt

```diff
@@ -43,4 +41,40 @@ class RevokeMembershipFlow(val membership : StateAndRef<Membership.State>) : Flo
 
         return finalisedTx
     }
+}
+
+/**
+ * This is a convenience flow that can be easily used from command line
+ *
+ * @param party whose membership state to be revoked
+ */
+@InitiatingFlow
+@StartableByRPC
+class RevokeMembershipForPartyFlow(val party : Party) : FlowLogic<SignedTransaction>() {
+
+    companion object {
+        object LOOKING_FOR_MEMBERSHIP_STATE : ProgressTracker.Step("Looking for party's membership state")
+        object REVOKING_THE_MEMBERSHIP_STATE : ProgressTracker.Step("Revoking the membership state")
+
+        fun tracker() = ProgressTracker(
+                LOOKING_FOR_MEMBERSHIP_STATE,
+                REVOKING_THE_MEMBERSHIP_STATE
+        )
+    }
+
+    override val progressTracker = tracker()
+
+    @Suspendable
+    override fun call() : SignedTransaction {
+        progressTracker.currentStep = LOOKING_FOR_MEMBERSHIP_STATE
+        val stateToActivate = findMembershipStateForParty(party)
+
+        progressTracker.currentStep = REVOKING_THE_MEMBERSHIP_STATE
+        return subFlow(RevokeMembershipFlow(stateToActivate))
+    }
+
+    private fun findMembershipStateForParty(party : Party) : StateAndRef<Membership.State> {
+        //@todo this could be made more effective and look for the Party's state in the vault
+        return serviceHub.vaultService.queryBy<Membership.State>().states.filter { it.state.data.member == party }.single()
```

Comment:
Similar to the above - lets throw a more meaningfull exception in the case if embership is missing

Reply:
ok